components/camel-salesforce/camel-salesforce-component/src/test/java/org/apache/camel/component/salesforce/internal/streaming/SubscriptionHelperManualIT.java

```diff
@@ -285,14 +293,70 @@ void shouldResubscribeOnConnectionFailures() throws InterruptedException {
         verify(consumer, atLeastOnce()).getTopicName();
         verifyNoMoreInteractions(consumer);
     }
 
+    @Test
+    void shouldResubscribeOnSubscriptionFailure() {
+        var subscribeAttempts = new AtomicInteger(0);
+        subscription.start();
+        subscription.client.getChannel("/meta/subscribe").addListener(
+                (MessageListener) (clientSessionChannel, message) -> subscribeAttempts.incrementAndGet());
+        var endpoint = mock(SalesforceEndpoint.class, "shouldResubscribeOnSubscriptionFailure:endpoint");
+        when(endpoint.getConfiguration()).thenReturn(config);
+        when(endpoint.getComponent()).thenReturn(salesforce);
+        when(endpoint.getTopicName()).thenReturn("Contact");
+        var consumer = toUnsubscribe = mock(StreamingApiConsumer.class, "shouldResubscribeOnSubscriptionFailure:consumer");
+        when(consumer.getTopicName()).thenReturn("Contact");
+        when(consumer.getEndpoint()).thenReturn(endpoint);
+
+        server.replyTo("POST", "/cometd/" + SalesforceEndpointConfig.DEFAULT_VERSION + "/subscribe",
+                s -> s.contains("/topic/Contact"),
+                """
+                        [
+                          {
+                            "clientId": "5ra4927ikfky6cb12juthkpofeu8",
+                            "channel": "/meta/subscribe",
+                            "id": "$id",
+                            "subscription": "/topic/Contact",
+                            "successful": false
+                          }
+                        ]""");
+        subscription.subscribe("Contact", consumer);
+        await().atMost(10, SECONDS).until(() -> subscribeAttempts.get() == 1);
+        assertThat(subscription.client.getChannel("/topic/Contact").getSubscribers(), hasSize(0));
+
+        server.reset();
+        server.replyTo("POST", "/cometd/" + SalesforceEndpointConfig.DEFAULT_VERSION + "/subscribe",
+                s -> s.contains("/topic/Contact"),
+                """
+                        [
+                          {
+                            "clientId": "5ra4927ikfky6cb12juthkpofeu8",
+                            "channel": "/meta/subscribe",
+                            "id": "$id",
+                            "subscription": "/topic/Contact",
+                            "successful": true
+                          }
+                        ]""");
+        messages.add("""
+                [
+                  {
+                    "clientId": "5ra4927ikfky6cb12juthkpofeu8",
+                    "channel": "/meta/connect",
+                    "id": "$id",
+                    "successful": true
+                  }
+                ]""");
+        await().atMost(10, SECONDS).until(() -> subscribeAttempts.get() == 2);
+        assertThat(subscription.client.getChannel("/topic/Contact").getSubscribers(), hasSize(1));
+    }
+
     @Test
     void shouldResubscribeOnHelperRestart() {
```

Comment:
This test seems flaky. Fails for me when the whole test class is run, but succeeds if only shouldResubscribeOnHelperRestart is run.

Reply:
Good catch, @jeremyross. Fixed.